src/Http/Http/src/Timeouts/CancellationTokenLinker.cs

```diff
@@ -0,0 +1,18 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Internal;
+
+namespace Microsoft.AspNetCore.Http.Timeouts;
+
+internal class CancellationTokenLinker : ICancellationTokenLinker
+{
+    private readonly CancellationTokenSourcePool _ctsPool = new();
+
+    public CancellationTokenSource GetLinkedCancellatinTokeSource(CancellationToken originalToken, TimeSpan timeSpan)
+    {
+        var cancellationTokenSource = _ctsPool.Rent();
```

Comment:
The pool isn't doing anything right now since the token is never returned to it.

Reply:
I'm calling `RegisterForDispose` here. Will this return it when the request is over?
https://github.com/dotnet/aspnetcore/blob/3d11e638c07198457e1207d6bdc46b99b8969f24/src/Http/Http/src/Timeouts/RequestTimeoutsMiddleware.cs#L92

Reply:
No
https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs,2f3ee94b8918bff2,references

Reply:
Oh, I see! My bad! I'm passing the token to the linkedCTS not the timeout cts.
I've pushed another commit and dispose the timeout cts.